### DISCUSSION TITLE
issue: Weird behaviour on group radio (when clear selection if already exist on click)

### DISCUSSION THREAD

**Discussion by awaismehmood88** [react-hook-form/react-hook-form#8711] [Q&A] [answered]
### Version Number

7.15.3

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-outside-submit-forked-ywbt8t?file=/src/App.js

### Steps to reproduce

1. Go to codesandbox
2. click on any radio
3. then click on other radio btn
4. first radio will be selected when clicking on second 

### Expected behaviour

when click on one radio button it will checked
when click second times it will unchecked (it will set null value on change)

I am overriding onchange function on input field when current value and state value (field.value) is same I am setting value to null in event object before forwading it to controller onchange function.

Problem only show when using override functionality in onchange and setting event = {target: {value: null}} 
*Error cause on null value

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment, marked as answer]
when using a controller, you are doing a collection data job with it, so you should wrap it with a controlled component with a state host inside for your radios inputs. eg `<Controller render={() => <RadioGroup />} />` eg: https://codesandbox.io/s/usecontroller-own-state-wncet2?file=/src/App.tsx
